service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -82,6 +90,21 @@ public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody Callback
         return respond(caseDetails);
     }
 
+    @PostMapping("/validate/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleValidateMidEvent(
```

Comment:
I think in this instance it would be beneficial to leave TODO / FIXME / comment stating why we need to have two similar methods. It's confusing when looking at it.

Reply:
Hmm, what do you mean sorry @akomisarek? Could we not just have

```
@Postmapping("validate-supporting-evidence-document)"
public AboutToStartOrSubmitCallbackResponse handleValidateSupportingEvidenceDocument
```

Reply:
@akomisarek meant to say, now in this PR, PBA validation is done twice once old way and once in generic endpoint. So, I have put TODO to remove old PBA validation.